### ISSUE TITLE
When trying to select a node with a vector new elements are accidentally added

### ISSUE THREAD

**Issue by sylwiabr** [enso-org/enso#6529]
### What type of issue is this?

Permanent – Occurring repeatably

### Is this issue blocking you from using Enso?

- [ ] Yes, I can't use Enso because of this issue.

### Is this a regression?

- [X] Yes, previous version of Enso did not have this issue.

### What issue are you facing?

https://user-images.githubusercontent.com/12892578/235905933-fe258979-21ed-45cd-89e0-cd14354db5f0.mov

When trying to select a node with a vector, node is getting selected but also new element is added.

### Expected behaviour

Selecting nodes should be easy.

### Enso Version

nightly 03.05

### Browser or standalone distribution

Standalone distribution

### Browser Version or standalone distribution

standalone

### Operating System

MacOS

**Comment by wdanilo** [comment]
This is caused by mouse not always showing the `+` icon. The icon should be shown if you place the mouse between items.

**Comment by wdanilo** [comment]
Also, the new design has an icon always on the left side of the node:
<img width="467" alt="image" src="https://user-images.githubusercontent.com/1623053/235933036-f8f7c4a2-2d00-4fa0-a6b2-16d62caf32f5.png">

Until we do it, dragging nodes with only widgets would be hard, so we could implement the new node deign ASAP.

**Comment by farmaazon** [comment]
Currently, adding a new element is fired on every up event, but I think we should fire it only when the down event was also in the list editor (or even in the right gap).

To be done after #6549 as the architecture of event handling may be revised there.

**Comment by farmaazon** [comment]
After fixes in #6540 I cannot reproduce adding new element (it's added only when cursor is plus, but this is expected). Also selecting with box and ending selection between elements didn't add elements to list editor.

The node is often not selected, but that's another issue: https://github.com/enso-org/enso/issues/6677

@sylwiabr please check it also and if you reproduce, please reopen this issue.
